misk-zookeeper/src/main/kotlin/misk/zookeeper/ZookeeperCommonModule.kt

```diff
@@ -0,0 +1,45 @@
+package misk.zookeeper
+
+import com.google.common.util.concurrent.Service
+import misk.clustering.zookeeper.ZookeeperConfig
+import misk.inject.KAbstractModule
+import misk.inject.asSingleton
+import misk.inject.keyOf
+import org.apache.curator.ensemble.EnsembleProvider
+import org.apache.curator.framework.CuratorFramework
+import javax.inject.Provider
+import kotlin.reflect.KClass
+
+/**
+ * Binds a [CuratorFramework] for an application to use. A [Service] is also installed to manage
+ * the lifecycle of the [CuratorFramework].
+ *
+ * If an application needs to connect to multiple Zookeepers, an optional qualifier can be passed
+ * resulting in an annotated [CuratorFramework] binding.
+ *
+ * ```
+ * @Qualifier
+ * annotation class MyZk
+ *
+ * install(ZookeeperCommonModule(config, MyZk::class)
+ *
+ * @Inject @MyZk CuratorFramework
+ * ```
+ */
+class ZookeeperCommonModule(
```

Comment:
nit: could probably just call this the `CuratorModule`, since that's the intention